server/src/main/java/io/crate/analyze/relations/FullQualifiedNameFieldProvider.java

```diff
@@ -156,7 +158,11 @@ public Symbol resolveField(QualifiedName qualifiedName, @Nullable List<String> p
             RelationName relationName = new RelationName(schema, columnTableName);
             throw new RelationUnknown(relationName);
         }
-        RelationName relationName = sources.entrySet().iterator().next().getKey();
+        var source = sources.entrySet().iterator().next();
+        RelationName relationName = source.getKey();
+        if (source.getValue() instanceof TableFunctionRelation) {
+            throw columnUnknownFromTableFunctionException(columnIdent.sqlFqn(), relationName);
```

Comment:
Maybe we need to rethink the scoping system,  but we shouldn't need different kind of unknown column exceptions and components raising them shouldn't need to know which one of them to raise